packages/expo-camera/android/src/main/java/expo/modules/camera/next/ExpoCameraView.kt

```diff
@@ -0,0 +1,481 @@
+                }
+              )
+            }
+          }
+        }.also { recording ->
+          recording.mute(mute)
+        }
+    }
+      ?: promise.reject("E_RECORDING_FAILED", "Starting video recording failed - could not create video file.", null)
+  }
+
+  @SuppressLint("UnsafeOptInUsageError")
+  private fun createCamera() {
+    val providerFuture = ProcessCameraProvider.getInstance(context)
+
+    providerFuture.addListener(
+      {
+        val cameraProvider: ProcessCameraProvider = providerFuture.get()
+
+        val preview = Preview.Builder()
+          .build()
+          .also {
+            it.setSurfaceProvider(previewView.surfaceProvider)
+          }
+
+        val cameraSelector = lenFacing.mapToSelector()
+
+        imageCaptureUseCase = ImageCapture.Builder()
+          .build()
+
+        val cameraInfo = cameraProvider.availableCameraInfos.filter {
+          Camera2CameraInfo
+            .from(it)
+            .getCameraCharacteristic(CameraCharacteristics.LENS_FACING) == CameraMetadata.LENS_FACING_BACK
+        }
+
+        val videoCapture = createVideoCapture(cameraInfo)
+        imageAnalysisUseCase = createImageAnalyzer()
+
+        val useCases = UseCaseGroup.Builder().apply {
+          addUseCase(preview)
+          if (cameraMode == CameraMode.PICTURE) {
+            imageCaptureUseCase?.let {
+              addUseCase(it)
+            }
+            imageAnalysisUseCase?.let {
+              addUseCase(it)
+            }
+          } else {
+            addUseCase(videoCapture)
+          }
+        }.build()
+
+        try {
+          cameraProvider.unbindAll()
+          camera = cameraProvider.bindToLifecycle(currentActivity, cameraSelector, useCases)
+          camera?.let {
+            observeCameraState(it.cameraInfo)
+          }
+        } catch (e: Exception) {
+          onMountError(
+            CameraMountErrorEvent("Camera component could not be rendered - is there any other instance running?")
+          )
+        }
+      },
+      ContextCompat.getMainExecutor(context)
+    )
+  }
+
+  private fun createImageAnalyzer(): ImageAnalysis =
+    ImageAnalysis.Builder()
+      .setBackpressureStrategy(ImageAnalysis.STRATEGY_KEEP_ONLY_LATEST)
+      .setResolutionSelector(
+        ResolutionSelector.Builder()
+          .setAspectRatioStrategy(AspectRatioStrategy.RATIO_16_9_FALLBACK_AUTO_STRATEGY)
+          .build()
+      )
+      .build()
+      .also { analyzer ->
+        if (shouldScanBarCodes) {
+          analyzer.setAnalyzer(
+            ContextCompat.getMainExecutor(context),
+            BarcodeAnalyzer(lenFacing, barcodeFormats) {
+              onBarCodeScanned(it)
+            }
+          )
+        }
+      }
+
+  private fun createVideoCapture(info: List<CameraInfo>): VideoCapture<Recorder> {
+    val supportedQualities = QualitySelector.getSupportedQualities(info[0])
+
+    val filteredQualities = arrayListOf(Quality.UHD, Quality.FHD, Quality.HD, Quality.SD)
```

Comment:
It may be optional right now, but what about 8k videos? I know that `Quality` enum has two more values: `HIGHEST` and `LOWEST`. I would assume that if the camera can handle capturing in 8k, you would have to use `Quality.HIGHEST` to enable it.

Reply:
I need to look into how we can support selecting video quality some more. I believe we can let the user select what they would like the quality level to be and CameraX will pick the closest available on the current device. I'd assume UHD will be selected if the device is capable of 8k.